kubeless-rbac.jsonnet

```diff
@@ -5,11 +5,28 @@ local objectMeta = k.core.v1.objectMeta;
 
 local kubeless = import "kubeless.jsonnet";
 local controller_account = kubeless.controller_account;
-local controller_roles = [{
-  apiGroups: ["*"],
-  resources: ["services", "deployments", "functions", "configmaps", "pods", "replicasets"],
-  verbs: ["*"]
-}];
+local controller_roles = [
+  {
+    apiGroups: [""],
+    resources: ["services", "configmaps"],
+    verbs: ["create", "get", "delete", "list", "update", "patch"],
+  },
+  {
+    apiGroups: ["apps", "extensions"],
+    resources: ["deployments"],
+    verbs: ["create", "get", "delete", "list", "update", "patch"],
+  },
+  {
+    apiGroups: [""],
+    resources: ["pods"],
+    verbs: ["list", "delete"],
+  },
+  {
+    apiGroups: ["k8s.io"],
+    resources: ["functions"],
+    verbs: ["*"],
```

Comment:
The controller only needs to get+list functions, correct?

Reply:
@anguslees : thanks && almost there, needs to "watch" also ->
```
Failed to watch *spec.Function: User "system:serviceaccount:kubeless:controller-acct" cannot watch functions.k8s.io at the cluster scope. (get functions.k8s.io)
```